lib/ansible/modules/network/cloudengine/ce_vrf_interface.py

```diff
@@ -0,0 +1,517 @@
+    elif interface.upper().startswith('4X10GE'):
+        iftype = '4x10ge'
+    elif interface.upper().startswith('40GE'):
+        iftype = '40ge'
+    elif interface.upper().startswith('100GE'):
+        iftype = '100ge'
+    elif interface.upper().startswith('VLANIF'):
+        iftype = 'vlanif'
+    elif interface.upper().startswith('LOOPBACK'):
+        iftype = 'loopback'
+    elif interface.upper().startswith('METH'):
+        iftype = 'meth'
+    elif interface.upper().startswith('ETH-TRUNK'):
+        iftype = 'eth-trunk'
+    elif interface.upper().startswith('VBDIF'):
+        iftype = 'vbdif'
+    elif interface.upper().startswith('NVE'):
+        iftype = 'nve'
+    elif interface.upper().startswith('TUNNEL'):
+        iftype = 'tunnel'
+    elif interface.upper().startswith('ETHERNET'):
+        iftype = 'ethernet'
+    elif interface.upper().startswith('FCOE-PORT'):
+        iftype = 'fcoe-port'
+    elif interface.upper().startswith('FABRIC-PORT'):
+        iftype = 'fabric-port'
+    elif interface.upper().startswith('STACK-PORT'):
+        iftype = 'stack-Port'
+    elif interface.upper().startswith('NULL'):
+        iftype = 'null'
+    else:
+        return None
+
+    return iftype.lower()
+
+
+class VrfInterface(object):
+    """Manange vpn instance"""
+
+    def __init__(self, argument_spec):
+        self.spec = argument_spec
+        self.module = None
+        self.init_module()
+
+        # vpn instance info
+        self.vrf = self.module.params['vrf']
+        self.vpn_interface = self.module.params['vpn_interface']
+        self.vpn_interface = self.vpn_interface.upper().replace(' ', '')
+        self.state = self.module.params['state']
+        self.intf_info = dict()
+        self.intf_info['isL2SwitchPort'] = None
+        self.intf_info['vrfName'] = None
+        self.conf_exist = False
+
+        # state
+        self.changed = False
+        self.updates_cmd = list()
+        self.results = dict()
+        self.proposed = dict()
+        self.existing = dict()
+        self.end_state = dict()
+
+    def init_module(self):
+        """init_module"""
+
+        self.module = AnsibleModule(
+            argument_spec=self.spec, supports_check_mode=True)
+
+    def check_response(self, xml_str, xml_name):
+        """Check if response message is already succeed."""
+
+        if "<ok/>" not in xml_str:
+            self.module.fail_json(msg='Error: %s failed.' % xml_name)
+
+    def get_update_cmd(self):
+        """ get  updated command"""
+
+        if self.conf_exist:
+            return
+
+        if self.state == 'absent':
+            self.updates_cmd.append(
+                "undo ip binding vpn-instance %s" % self.vrf)
+            return
+
+        if self.vrf != self.intf_info['vrfName']:
+            self.updates_cmd.append("ip binding vpn-instance %s" % self.vrf)
+
+        return
+
+    def check_params(self):
+        """Check all input params"""
+
+        if not self.vrf or not self.vpn_interface:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
required_together
required_one_of
require_if
```